NETProvider/src/FirebirdSql.Data.FirebirdClient/Client/Common/XsqldaMarshaler.cs

```diff
@@ -72,13 +50,14 @@ public void CleanUpNativeData(ref IntPtr pNativeData)
 					IntPtr ptr1 = GetIntPtr(pNativeData, ComputeLength(i));
 
 					// Free	sqldata	and	sqlind pointers	if needed
-					XSQLVAR sqlvar = (XSQLVAR)Marshal.PtrToStructure(ptr1, typeof(XSQLVAR));
+					XSQLVAREmpty sqlvar = (XSQLVAREmpty)Marshal.PtrToStructure(ptr1, typeof(XSQLVAREmpty));
```

Comment:
In this version there's no allocation in `XSQLVAR` (the first version had `new byte[32]` there, but no more - now it's handled by the `BinaryReader`).